sr_edc_ethercat_drivers/include/sr_edc_ethercat_drivers/sr10.h

```diff
@@ -0,0 +1,215 @@
+/* 
+* Copyright 2023 Shadow Robot Company Ltd.
+*
+* This program is free software: you can redistribute it and/or modify it
+* under the terms of the GNU General Public License as published by the Free
+* Software Foundation version 2 of the License.
+*
+* This program is distributed in the hope that it will be useful, but WITHOUT
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+* more details.
+*
+* You should have received a copy of the GNU General Public License along
+* with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+  * @file   sr10.h
+  * @author Yann Sionneau <yann.sionneau@gmail.com>, Hugo Elias <hugo@shadowrobot.com>,
+  *         Ugo Cupcic <ugo@shadowrobot.com>, Toni Oliver <toni@shadowrobot.com>,
+  *         Dan Greenwald <dg@shadowrobot.com>, Rodrigo Zenha <rodrigo@shadowrobot.com>
+  *         contact <software@shadowrobot.com>
+  * @brief This is a ROS driver for Shadow Robot #10 EtherCAT product ID
+  */
+
+#ifndef SR_EDC_ETHERCAT_DRIVERS_SR10_H
+#define SR_EDC_ETHERCAT_DRIVERS_SR10_H
+
+#include <ros_ethercat_hardware/ethercat_hardware.h>
+#include <ros_ethercat_model/robot_state.hpp>
+#include <sr_edc_ethercat_drivers/sr_edc.h>
+#include <realtime_tools/realtime_publisher.h>
+#include <std_msgs/Int16.h>
+#include <std_msgs/Float64MultiArray.h>
+#include <sr_robot_msgs/SetImuScale.h>
+#include <sr_robot_msgs/SimpleMotorFlasher.h>
+#include <pthread.h>
+#include <bfd.h>
+#include <boost/smart_ptr.hpp>
```

Comment:
Is there any specific reason why we use the boost library instead of the standard library?

Reply:
I kept it since we are using it throughout the whole codebase of ros-interface (e.g. previous FW). It seems that historically boost has implemented smart pointers first (which only became available to std from C11), so I'm guessing that is why we started using them in the first place. I'll attempt changing them for the files of this new version (the package would still depend on boost), but later a new story should be created to refactor the codebase

Reply:
Changed (and tested) here: https://github.com/shadow-robot/sr-ros-interface-ethercat/pull/265/commits/1270cd812a2f10d072b0191c0b27eef69c7f50a1

Reply:
Changed (and tested) here: https://github.com/shadow-robot/sr-ros-interface-ethercat/pull/265/commits/1270cd812a2f10d072b0191c0b27eef69c7f50a1